File: 1627-last-moment-before-all-ants-fall-out-of-a-plank/1627-last-moment-before-all-ants-fall-out-of-a-plank.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        """
        Actually, I don't have to differenciate each ant, so I cannot tell
        1) ants meet and change others from 2) ants meet and go the same direction.
        """

        answer = []
        if left:
            answer.append(  max(left) )
        if right:
            answer.append(  n - min(right)  )
        return max(answer)

File: 1627-last-moment-before-all-ants-fall-out-of-a-plank/1627-last-moment-before-all-ants-fall-out-of-a-plank.java
class Solution {
	public int getLastMoment(int n, int[] left, int[] right) {
		var answer = new ArrayList<Integer>();

		answer.add(Arrays.stream(left).max().orElse(-1));
		answer.add(n - Arrays.stream(right).min().orElse(n));

		return answer
			.stream()
			.mapToInt(Integer::valueOf)
			.max()
			.orElseThrow();
	}
}